### DISCUSSION TITLE
Difference between seen_at and read_at?

### DISCUSSION THREAD

**Discussion by ibrahima** [excid3/noticed#497] [Q&A]
Hi, I've got a simple question. What's the intended difference between `seen_at` and `read_at`? The readme mostly mentions `read`/`mark_as_read`. Is it like, one is for notifications having been delivered? But that doesn't quite make sense to me as `seen`. Would appreciate any clarification or examples of how people are using this, thanks!

On a tangentially related note, it seems like it'd be good to have indexes on these fields, because it seems like getting unread notifications would be a common operation.

**Comment by marckohlbrugge** [comment]
I had the same question. There's nothing in the `noticed` codebase that explains the difference or even functionally treat them differently.

However, the [Jumpstart Pro](https://jumpstartrails.com) template does use both columns for slightly different purposes. I believe the `seen_at` is used to indicate whether a red badge has been shown to indicate to the user there's a new notification. Whereas the `read_at` is used to track whether they actually saw the contents of the notification.

I think the Noticed code was extracted from Jumpstart which is why it ended up in the gem.
